src/test/java/org/broadinstitute/consent/http/mail/freemarker/FreeMarkerTemplateHelperTest.java

```diff
@@ -121,6 +121,55 @@ public void testGetClosedDatasetElectionsTemplate() throws Exception {
         assertTrue(parsedTemplate.getElementById("userName").text().equals("Hello Admin!"));
     }
 
+    @Test
+    public void testGetNewResearcherLibraryRequestTemplate() throws Exception {
+        Writer template = helper.getNewResearcherLibraryRequestTemplate("John Doe", "http://localhost:8000/#/");
+        String templateString = template.toString();
+
+        assertEquals(
```

Comment:
Ok so I personally like that you have the entire email here. Understandably, the code gets bloated. However, it's very easy to check how it actually renders. If I want to know what the email looks like, I just check this. Otherwise, to test styling, links, etc, I would have to manually put in the replacement variables, which allows for human error and slower testing, and if the email accidentally changes and style breaks, we don't know that. Having the code this way makes it very easy to see how the email renders, and ensure that styling never breaks (unless intentional). Perhaps, we could clean this up by putting this in its own file.

Reply:
> if the email accidentally changes and style breaks, we don't know that

We'll still not know if styles break - in practice, any changes to the template file will be copied here just to make the tests pass. Philosophically, we're going beyond the unit test by expecting things like styling to "work", so to speak. What I think you want is something that tests `new-researcher-library-request.html` and only the content of that file, which is not in the scope of this unit test. The boundary around `getNewResearcherLibraryRequestTemplate` is in two variables, the name and the url, so that's what a unit test around this method should be testing.

Reply:
>  Philosophically, we're going beyond the unit test by expecting things like styling to "work", so to speak

Sure, but we would still be checking the content of the file right? We would still be rendering the HTML and checking that the content is replaced properly. If it is in the scope of the test to check the raw content of the HTML, I don't see why styling isn't also included with that. I would argue that styling is an aspect of rendering, and here are the tests to ensure proper rendering, styling and replacement both included.

Either way, we don't have any other place to check that... I am ok putting this elsewhere if you prefer. 

As far as copying and pasting, I think that forcing the programmer to copy over the template and change the variables forces them to actively make the decision to not check that the HTML renders properly, which I think would be strange. As opposed to, the template changes, and then the person testing has to manually make those changes to test the final rendered email, if they even notice the diff changes, which is subject to a multitude of human errors. Basically, I feel like there has to be some place that we can make sure the final email is all good, and I don't see where that would be other than here. And I don't feel comfortable _not_ having that for emails going forward, because it adds friction to testing something that actual users in production will see.

Idk though, maybe I'm off base here. If you feel strongly, I can change it to match other tests in this class.

Reply:
I'm not against testing the content - I'm advocating for proper testing boundaries around methods. What I think you're doing here is conflating the need for testing style with the right place to make that test. The unit test of `getNewResearcherLibraryRequestTemplate` doesn't have any concerns about the styling of the template. By adding additional styling logic to a test for **_that_** method makes the test do double duty which is what I'm trying to avoid. 

My suggestion is to isolate the testing for this method to what the method cares about. Secondly, if we really want to test the styling, then create a separate test that reads `new-researcher-library-request.html`, does variable interpolation on it, and ensures that it works the way we want it to.